src/CredentialProvider.ts

```diff
@@ -1,43 +1,45 @@
-import { Credentials } from './Credentials.js'
+import { Credentials } from './Credentials.ts'
 
 export class CredentialProvider {
-  constructor({ accessKey, secretKey, sessionToken }) {
+  private credentials: Credentials
+
+  constructor({ accessKey, secretKey, sessionToken }: { accessKey: string; secretKey: string; sessionToken: string }) {
     this.credentials = new Credentials({
       accessKey,
       secretKey,
       sessionToken,
     })
   }
 
-  getCredentials() {
+  async getCredentials(): Promise<Credentials | null> {
```

Comment:
```suggestion
  getCredentials(): Promise<Credentials | null> {
```

No need to use `async` keyword if we're not using any `await`. Usually this would fall into this specific eslint rule: https://eslint.org/docs/latest/rules/require-await

Reply:
Yes but it's necessary here, subclass may need to be an async function, like out `AssumeRoleProvider`

Reply:
`AssumeRoleProvider` may do http request if its cred is expired, so this method need to be async function.